src/main/kotlin/com/avail/interpreter/levelTwo/operation/L2_REENTER_L1_CHUNK_FROM_INTERRUPT.kt

```diff
@@ -0,0 +1,119 @@
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.avail.interpreter.levelTwo.operation
+
+import com.avail.descriptor.functions.A_Continuation
+import com.avail.interpreter.execution.Interpreter
+import com.avail.interpreter.execution.Interpreter.Companion.log
+import com.avail.interpreter.levelTwo.L2Instruction
+import com.avail.interpreter.levelTwo.L2Operation
+import com.avail.optimizer.jvm.CheckedMethod
+import com.avail.optimizer.jvm.JVMTranslator
+import com.avail.optimizer.jvm.ReferencedInGeneratedCode
+import com.avail.utility.Nulls
+import org.objectweb.asm.MethodVisitor
+import java.util.logging.Level
+
+/**
+ * This is the first instruction of the L1 interpreter's on-ramp for resuming
+ * after an interrupt.  The reified [A_Continuation] that was captured
+ * (and is now being resumed) pointed to this [L2Instruction].  That
+ * continuation is current in the [Interpreter.getReifiedContinuation].
+ * Pop it from that continuation chain, create suitable pointer and integer
+ * registers as expected by [L2_INTERPRET_LEVEL_ONE], then explode the
+ * continuation's slots into those registers.  The [Interpreter.function]
+ * should also have already been set up to agree with the continuation's
+ * function.
+ *
+ * @author Mark van Gulik &lt;mark@availlang.org&gt;
+ * @author Todd L Smith &lt;todd@availlang.org&gt;
+ */
+object L2_REENTER_L1_CHUNK_FROM_INTERRUPT : L2Operation()
+{
+	override fun hasSideEffect(): Boolean = true
+
+	override fun isEntryPoint(instruction: L2Instruction): Boolean = true
+
+	override fun translateToJVM(
+		translator: JVMTranslator,
+		method: MethodVisitor,
+		instruction: L2Instruction)
+	{
+		// :: L2_REENTER_L1_CHUNK_FROM_INTERRUPT.reenter();
+		translator.loadInterpreter(method)
+		reenterMethod.generateCall(method)
+	}
+
+	/**
+	 * Reenter from an interrupt.
+	 *
+	 * @param interpreter
+	 *   The [Interpreter].
+	 */
+	@JvmStatic
+	@ReferencedInGeneratedCode
+	fun reenter(interpreter: Interpreter)
+	{
+		val continuation: A_Continuation =
+			Nulls.stripNull(interpreter.getReifiedContinuation())
+		interpreter.setReifiedContinuation(continuation.caller())
+		if (Interpreter.debugL1)
+		{
+			log(
+				Interpreter.loggerDebugL1,
+				Level.FINER,
+				"{0}Reenter L1 from interrupt",
+				interpreter.debugModeString)
+		}
+		val function = Nulls.stripNull(interpreter.function)
+		assert(function === continuation.function())
+		val numSlots = continuation.numSlots()
+		// Should agree with L2_PREPARE_NEW_FRAME_FOR_L1.
+		val stepper = interpreter.levelOneStepper
+		stepper.pointers = arrayOfNulls(numSlots + 1)
+		var dest = 1
+		for (i in 1 .. numSlots)
```

Comment:
Huh, I guess dest and i are always the same, so we may as well drop dest entirely.